docs: private Git repo usage in case of custom TCP port and repo path (#12793)
argoproj/argo-cd: docs/user-guide/private-repositories.md

!!!note 
    When your SSH repository is served from a non-standard port, you have to use `ssh://`-style URLs to specify your repository. The scp-style `git@yourgit.com:yourrepo` URLs do **not** support port specification, and will treat any port number as part of the repository's path.
    For example, the `gituser` user connecting to the `gitserver.example.com` server on port TCP/2222 for the repository located in the `/home/mygitrepo/` folder on the server:
    ```
    ssh://gituser@gitserver.example.com:2222/home/mygitrepo/
    ```

>     If the repository does not have a `master` branch with an initial commit, the connection status will be `Failed`.

Author: So here is what I tested:

1.) created my own Git repo on my server: `git init --bare ./repo_name` 
2.) pushed content from my local `main` to new remote `main` branch: `git push origin main`
3.) Added new repo to argo: `argocd repo add ssh://user_name@IP:2234/home/repo_name/ --ssh-private-key-path ~/.ssh/argocd --insecure-ignore-host-key --type git`
4.) Got the following error from argo:  Failed  Unable to connect to repository: rpc error: code = Unknown desc = error testing repository connectivity: Unable to resolve 'HEAD' to a commit SHA

So in this case I have `main` branch in my own Git repo and this looks like my default repo with several commit in it:

git branch -r
  origin/main
git push
Everything up-to-date

It might be `git init` specific, but you need to use `-b` to create `main` branch on Git server side, because by default it will be `master`. I can check it on Git server in the `./repo_name/HEAD` file which has following content in my case: `ref: refs/heads/master`. So it means I need to create my remote branch like this to prepare it for `main` branch really: `git init -b main --bare ./repo_name` or use the default `master` branch which is the default:

git init -h

       -b <branch-name>, --initial-branch=<branch-name>
           Use the specified name for the initial branch in the newly created repository. If not
           specified, fall back to the default name (currently master, but this is subject to
           change in the future; the name can be customized via the init.defaultBranch
           configuration variable).

So what do you recommend here? Should we highlight it as `git init` specific behavior only? I haven't tested other Git servers, probably those can create `main` branch on server side during repo initialization and switch HEAD to that.

>     Here is a working example with `gituser` user to access `gitserver.example.com` Git server on TCP/2222 port if Git repository is located in `/home/mygitrepo/` folder on the server side:
>     ```
>     ssh://gituser@gitserver.example.com:2222/home/mygitrepo/
>     ```
>     Be sure about your repo has a `master` branch with initial commit in it otherwise connetion status will be *Failed*.

Reviewer: Thanks for the contribution!  Thoughts on these wording tweaks?
```suggestion
    For example, the `gituser` user connecting to the `gitserver.example.com` server on port TCP/2222 for the repository located in the `/home/mygitrepo/` folder on the server:
    ```
    ssh://gituser@gitserver.example.com:2222/home/mygitrepo/
    ```
    If the repository does not have a `master` branch with an initial commit, the connection status will be `Failed`.
```

Author: It looks better now, thanks for the feedback.
